app/sql.js

```diff
@@ -320,11 +381,106 @@ async function updateToSchemaVersion4(currentVersion, instance) {
   console.log('updateToSchemaVersion4: success!');
 }
 
+async function updateToSchemaVersion6(currentVersion, instance) {
+  if (currentVersion >= 6) {
+    return;
+  }
+  console.log('updateToSchemaVersion6: starting...');
+  await instance.run('BEGIN TRANSACTION;');
+
+  // key-value, ids are strings, one extra column
+  await instance.run(
+    `CREATE TABLE sessions(
+      id STRING PRIMARY KEY ASC,
+      number STRING,
+      json TEXT
+    );`
+  );
+
+  await instance.run(`CREATE INDEX sessions_number ON sessions (
+    number
+  ) WHERE number IS NOT NULL;`);
+
+  // key-value, ids are strings
+  await instance.run(
+    `CREATE TABLE groups(
+      id STRING PRIMARY KEY ASC,
+      json TEXT
+    );`
+  );
+  await instance.run(
+    `CREATE TABLE identityKeys(
+      id STRING PRIMARY KEY ASC,
+      json TEXT
+    );`
+  );
+  await instance.run(
+    `CREATE TABLE items(
+      id STRING PRIMARY KEY ASC,
+      json TEXT
+    );`
+  );
+
+  // key-value, ids are integers
+  await instance.run(
+    `CREATE TABLE preKeys(
+      id INTEGER PRIMARY KEY ASC,
+      json TEXT
+    );`
+  );
+  await instance.run(
+    `CREATE TABLE signedPreKeys(
+      id INTEGER PRIMARY KEY ASC,
+      json TEXT
+    );`
+  );
+
+  await instance.run('PRAGMA schema_version = 6;');
+  await instance.run('COMMIT TRANSACTION;');
+  console.log('updateToSchemaVersion6: success!');
+}
+
+async function updateToSchemaVersion7(currentVersion, instance) {
+  if (currentVersion >= 7) {
+    return;
+  }
+  console.log('updateToSchemaVersion7: starting...');
+  await instance.run('BEGIN TRANSACTION;');
+
+  // Add recipient row
+  await instance.run('ALTER TABLE preKeys ADD recipient STRING;');
+
+  await instance.run(
+    `CREATE TABLE contactPreKeys(
+      id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+      identityKeyString VARCHAR(255),
+      keyId INTEGER,
+      json TEXT
+    );`
+  );
+
+  await instance.run(
+    `CREATE TABLE contactSignedPreKeys(
+      id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+      identityKeyString VARCHAR(255),
+      keyId INTEGER,
+      json TEXT
+    );`
+  );
+
+  await instance.run('PRAGMA schema_version = 7;');
+  await instance.run('COMMIT TRANSACTION;');
+  console.log('updateToSchemaVersion7: success!');
+}
+
 const SCHEMA_VERSIONS = [
   updateToSchemaVersion1,
   updateToSchemaVersion2,
   updateToSchemaVersion3,
   updateToSchemaVersion4,
+  // version 5 was dropped
+  updateToSchemaVersion6,
+  updateToSchemaVersion7, // Loki schema
```

Comment:
As discussed offline, we might benefit from shoehorning our changes into Version 6 to prevent conflicts with any upcoming version 7 from upstream/Signal.